gio-subclass/tests/simple_action.rs

```diff
@@ -0,0 +1,386 @@
+// Copyright (C) 2018 Sebastian Dröge <sebastian@centricular.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::mem;
+use std::ptr;
+use std::rc::Rc;
+use std::cell::RefCell;
+use std::sync::{Once, ONCE_INIT};
+
+extern crate glib_sys as glib_ffi;
+extern crate gobject_sys as gobject_ffi;
+
+#[macro_use]
+extern crate glib;
+use glib::prelude::*;
+use glib::translate::*;
+extern crate gio;
+use gio::prelude::*;
+extern crate gio_sys as gio_ffi;
+
+use glib::value::AnyValue;
+
+#[macro_use]
+extern crate gobject_subclass;
+use gobject_subclass::object::*;
+
+#[macro_use]
+extern crate gio_subclass;
+use gio_subclass::action::*;
+
+mod imp {
+    use super::*;
+    use std::cell::RefCell;
+
+    static PROPERTIES: [Property; 5] = [
+        Property::String(
+            "name",
+            "Action Name",
+            "The name used to invoke the action",
+            None,
+            PropertyMutability::ReadWrite // | PropertyMutability::ConstructOnly
+                // | PropertyMutability::StaticStrings,
+        ),
+        Property::Boxed(
+            "parameter-type",
+            "Parameter Type",
+            "The type of GVariant passed to activate()",
+            glib::VariantType::static_type,
+            PropertyMutability::ReadWrite// |
+            // PropertyMutability::ConstructOnly |
+            // PropertyMutability::StaticStrings,
+        ),
+        Property::Boolean(
+            "enabled",
+            "Enabled",
+            "If the action can be activated",
+            true,
+            PropertyMutability::ReadWrite
+             // | PropertyMutability::StaticStrings,
+        ),
+        Property::Boxed(
+            "state-type",
+            "State Type",
+            "The type of the state kept by the action",
+            glib::VariantType::static_type,
+            PropertyMutability::ReadWrite //|
+            // PropertyMutability::ConstructOnly |
```

Comment:
This needs some fixing

Reply:
I know. I'm going to leave that for now and address that in a separate PR.